### PR TITLE
create customers page with validation and search

### PR DESCRIPTION
**Pull request by 1Artemida**

### CODE REVIEW

File: client/src/pages/CustomersListPage/components/CustomersListPage.jsx [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  const [customersPerPage] = useState(7);
+  const [searchUsers, setSearchUsers] = useState([]);
+  const [totalPages, setTotalPages] = useState(0);
+  
+  const lastCustomers = currentPage * customersPerPage;
+  const firstCustomers = lastCustomers - customersPerPage;
+
+  useEffect(async() => {
+    
+    const response = await dispatch(fetchCustomersList({}));
```
Comment (MattMsh) [inline client/src/pages/CustomersListPage/components/CustomersListPage.jsx]: Is empty object as parameter required?

File: client/src/pages/CustomersListPage/components/CustomersListPage.jsx [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+  const setPaginateOptions = (event, pageNumber) => {
+    const lastCustomers = pageNumber * customersPerPage;
+    const firstCustomers = lastCustomers - customersPerPage;
+    setCurrentPage(pageNumber);
+    setSearchUsers(customersList.slice(firstCustomers, lastCustomers));
+  };
+  
+  const searchCustomersList = (value) => {
+    const lastCustomers = 1 * customersPerPage;
```
Comment (MattMsh) [inline client/src/pages/CustomersListPage/components/CustomersListPage.jsx:62]: What is this action sense? (lane 62)

File: client/src/pages/CustomersListPage/components/CustomersListPagination/CustomersListPagination.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  
+  return (
+    <Pagination
+      page={Number(currentPage)}
+      count={totalPages} 
+      onChange={setPaginateOptions}
+      shape='rounded'
+    />
+  );
+
```
Comment (Vadym342) [inline client/src/pages/CustomersListPage/components/CustomersListPagination/CustomersListPagination.js]: )))

File: client/src/routes/route.js [hunk 4]
```diff
@@ -32,6 +33,11 @@ const Routes = () => {
         path={`${routes.AuthRoutes.pathToProject}/:id`}
         component={ProjectPage}
       />
+      <AuthRoute
+        exact
+        path={`${routes.AuthRoutes.pathToCustomers}`}
```
Comment (Vadym342) [inline client/src/routes/route.js]: Why it is used here `${} `?

**Review by Vadym342 — APPROVED** [review]

File: client/src/pages/CustomersListPage/components/CustomersListPage.jsx (hunk 1 above)
Comment (1Artemida) [reply client/src/pages/CustomersListPage/components/CustomersListPage.jsx]: fixed

**Review by MattMsh — APPROVED** [review]

**Pull request merged**